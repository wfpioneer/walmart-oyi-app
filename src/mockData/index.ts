
import itemDetails from './getItemDetails';


export const getMockItemDetails = (upc: string) => {
  // @ts-ignore
  return itemDetails[upc] || {};
};
