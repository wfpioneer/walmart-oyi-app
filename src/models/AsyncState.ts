export interface AsyncState{
  isWaiting: boolean;
  value: any;
  error: any;
  result: any;
}
