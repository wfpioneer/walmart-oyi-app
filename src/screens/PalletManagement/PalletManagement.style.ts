import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

export default StyleSheet.create({
  scanContainer: {
    flexDirection: 'column',
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%'
  },
  textInput: {
    paddingLeft: 10,
    color: COLOR.BLACK,
    borderWidth: 1
  },
  scanText: {
    paddingTop: 30
  },
  orText: {
    paddingTop: 20
  },
  textView: {
    width: '60%',
    paddingTop: 20
  },
  activityIndicator: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
});
