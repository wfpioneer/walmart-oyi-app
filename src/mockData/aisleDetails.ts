import { AisleItem } from '../models/LocationItem';

export const mockAisles: AisleItem[] = [
  {
    aisleId: 1,
    aisleName: 'Aisle 1',
    sectionCount: 30
  },
  {
    aisleId: 2,
    aisleName: 'Aisle 2',
    sectionCount: 0
  },
  {
    aisleId: 3,
    aisleName: 'Aisle 3',
    sectionCount: 2
  },
  {
    aisleId: 4,
    aisleName: 'Aisle 4',
    sectionCount: 2
  },
  {
    aisleId: 5,
    aisleName: 'Aisle 5',
    sectionCount: 30
  },
  {
    aisleId: 6,
    aisleName: 'Aisle 6',
    sectionCount: 0
  }
];
