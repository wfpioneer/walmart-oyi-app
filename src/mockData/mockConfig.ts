import { Configurations, area } from '../models/User';

export const mockAreas: area[] = [
  {
    area: 'CENTER',
    categories: [5, 7, 9, 11, 13, 16]
  },
  {
    area: 'GM',
    categories: [21, 23, 24, 28, 31]
  },
  {
    area: 'Fresh',
    categories: [41, 42, 43, 45, 47, 48]
  }
];

export const mockConfig: Configurations = {
  locationManagement: false,
  locationManagementEdit: false,
  palletManagement: false,
  settingsTool: false,
  printingUpdate: false,
  binning: false,
  palletExpiration: false,
  backupCategories: '',
  picking: false,
  areas: mockAreas
};
