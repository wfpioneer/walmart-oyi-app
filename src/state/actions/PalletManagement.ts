import { Pallet, PalletItem } from '../../models/PalletManagementTypes';

export const SHOW_MANAGE_PALLET_MENU = 'PALLET_MANAGEMENT/SHOW_MANAGE_PALLET_MENU';
export const SETUP_PALLET = 'PALLET_MANAGEMENT/SETUP_PALLET';
export const CLEAR_PALLET_MANAGEMENT = 'PALLET_MANAGEMENT/CLEAR_PALLET_MANAGEMENT';
export const ADD_PALLET = 'PALLET_MANAGEMENT/ADD_PALLET';

export const showManagePalletMenu = (show: boolean) => ({
  type: SHOW_MANAGE_PALLET_MENU,
  payload: show
} as const);

export const setupPallet = (pallet: Pallet) => ({
  type: SETUP_PALLET,
  payload: pallet
} as const);

export const clearPalletManagement = () => ({
  type: CLEAR_PALLET_MANAGEMENT
} as const);

export const addItemToPallet = (palletItem: PalletItem) => ({
  type: ADD_PALLET,
  payload: palletItem
} as const);

export type Actions =
  ReturnType<typeof showManagePalletMenu>
  | ReturnType<typeof setupPallet>
  | ReturnType<typeof clearPalletManagement>
  | ReturnType<typeof addItemToPallet>;
