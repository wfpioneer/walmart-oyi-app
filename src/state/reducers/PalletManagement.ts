import {
  ADD_COMBINE_PALLET,
  ADD_ITEM,
  Actions,
  CLEAR_COMBINE_PALLET,
  CLEAR_PALLET_MANAGEMENT,
  CREATE_PALLET,
  DELETE_ITEM,
  REMOVE_COMBINE_PALLET,
  REMOVE_ITEM,
  RESET_PALLET,
  SETUP_PALLET,
  SET_ITEM_NEW_QUANTITY,
  SET_ITEM_QUANTITY,
  SET_PALLET_NEW_EXPIRY,
  SET_PERISHABLE_CATEGORIES,
  SHOW_MANAGE_PALLET_MENU,
  UPDATE_PALLET,
  UPDATE_PALLET_EXPIRATION_DATE
} from '../actions/PalletManagement';
import { CombinePallet, PalletInfo } from '../../models/PalletManagementTypes';
import { PalletItem } from '../../models/PalletItem';

interface PalletManagementState {
  managePalletMenu: boolean;
  palletInfo: PalletInfo;
  items: PalletItem[];
  combinePallets: CombinePallet[];
  perishableCategories: number[];
  createPallet: boolean;
}

const initialState: PalletManagementState = {
  managePalletMenu: false,
  palletInfo: {
    id: '0',
    createDate: undefined,
    expirationDate: undefined
  },
  items: [],
  combinePallets: [],
  perishableCategories: [],
  createPallet: false
};

export const PalletManagement = (state = initialState, action: Actions): PalletManagementState => {
  switch (action.type) {
    case SHOW_MANAGE_PALLET_MENU:
      return {
        ...state,
        managePalletMenu: action.payload
      };
    case SETUP_PALLET: {
      const { perishableCategories } = state;
      return {
        ...initialState,
        ...action.payload,
        perishableCategories
      }; }
    case ADD_COMBINE_PALLET:
      return {
        ...state,
        combinePallets: [...state.combinePallets, action.payload]
      };
    case CLEAR_COMBINE_PALLET:
      return {
        ...state,
        combinePallets: []
      };
    case REMOVE_COMBINE_PALLET: {
      const deleteIndex = state.combinePallets.findIndex(item => item.palletId === action.payload);
      const updatedPallets = [
        ...state.combinePallets.slice(0, deleteIndex),
        ...state.combinePallets.slice(deleteIndex + 1)
      ];
      return {
        ...state,
        combinePallets: updatedPallets
      }; }
    case ADD_ITEM:
      return {
        ...state,
        items: [...state.items, action.payload]
      };
    case RESET_PALLET: {
      const resetItems = state.items.map(item => {
        item.deleted = false;
        return item;
      });
      return {
        ...state,
        items: resetItems
      };
    }
    case DELETE_ITEM: {
      const updatedItems = state.items.map(item => {
        if (item.itemNbr.toString() === action.payload.itemNbr) {
          item.deleted = true;
          return item;
        }
        return item;
      });
      return {
        ...state,
        items: updatedItems
      };
    }
    case UPDATE_PALLET:
      return {
        ...state,
        items: [...action.payload]
      };
    case CLEAR_PALLET_MANAGEMENT:
      return initialState;
    case SET_ITEM_NEW_QUANTITY: {
      const newItems = state.items.map(item => {
        if (item.itemNbr.toString() === action.payload.itemNbr) {
          item.newQuantity = action.payload.newQuantity;
          return item;
        }
        return item;
      });
      return {
        ...state,
        items: newItems
      };
    }
    case SET_ITEM_QUANTITY: {
      const newItems = state.items.map(item => {
        if (item.itemNbr.toString() === action.payload.itemNbr && item.newQuantity) {
          item.quantity = item.newQuantity;
          return item;
        }
        return item;
      });
      return {
        ...state,
        items: newItems
      };
    }
    case REMOVE_ITEM: {
      const { items: palletItems } = state;
      const itemIdx = palletItems.findIndex(item => item.itemNbr.toString() === action.payload.itemNbr);
      const updatedItems = [...palletItems.slice(0, itemIdx), ...palletItems.slice(itemIdx + 1)];
      return {
        ...state,
        items: updatedItems
      };
    }
    case SET_PALLET_NEW_EXPIRY: {
      const { palletInfo } = state;
      const updatedPalletInfo: PalletInfo = {
        ...palletInfo,
        newExpirationDate: action.payload
      };
      return {
        ...state, palletInfo: updatedPalletInfo
      };
    }
    case SET_PERISHABLE_CATEGORIES: {
      const { perishableCategories } = action.payload;
      return {
        ...state,
        perishableCategories
      };
    }
    case UPDATE_PALLET_EXPIRATION_DATE: {
      const { palletInfo } = state;
      const updatedPalletInfo: PalletInfo = {
        ...palletInfo,
        expirationDate: palletInfo.newExpirationDate,
        newExpirationDate: undefined
      };
      return {
        ...state,
        palletInfo: updatedPalletInfo
      };
    }
    case CREATE_PALLET: {
      const { perishableCategories } = state;
      return {
        ...initialState,
        perishableCategories,
        createPallet: true
      };
    }
    default:
      return {
        ...state
      };
  }
};
