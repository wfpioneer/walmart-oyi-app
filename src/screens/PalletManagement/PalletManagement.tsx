import React, {
  Dispatch, EffectCallback, useEffect, useState
} from 'react';
import {
  ActivityIndicator,
  EmitterSubscription,
  Text,
  TextInput,
  TouchableOpacity,
  View
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {
  NavigationProp,
  RouteProp,
  useNavigation,
  useRoute
} from '@react-navigation/native';
import { useDispatch } from 'react-redux';
import Toast from 'react-native-toast-message';
import COLOR from '../../themes/Color';
import styles from './PalletManagement.style';
import { strings } from '../../locales';
import { barcodeEmitter, openCamera } from '../../utils/scannerUtils';
import { validateSession } from '../../utils/sessionTimeout';
import { getPalletInfo, getPalletConfig } from '../../state/actions/saga';
import { AsyncState } from '../../models/AsyncState';
import { useTypedSelector } from '../../state/reducers/RootReducer';
import { trackEvent } from '../../utils/AppCenterTool';
import { setPerishableCategories, setupPallet } from '../../state/actions/PalletManagement';
import {GET_PALLET_CONFIG, GET_PALLET_INFO} from '../../state/actions/asyncAPI';
import { Pallet, PalletItem } from '../../models/PalletManagementTypes';
import { Configurations } from '../../models/User';

interface PalletManagementProps {
  useEffectHook: (effect: EffectCallback, deps?: ReadonlyArray<any>) => void;
  searchText: string;
  setSearchText: React.Dispatch<React.SetStateAction<string>>;
  configComplete: boolean;
  setConfigComplete: React.Dispatch<React.SetStateAction<boolean>>;
  getInfoComplete: boolean;
  setGetInfoComplete: React.Dispatch<React.SetStateAction<boolean>>;
  navigation: NavigationProp<any>;
  route: RouteProp<any, string>;
  dispatch: Dispatch<any>;
  getPalletInfoApi: AsyncState;
  getPalletConfigApi: AsyncState;
  userConfig: Configurations
}
const palletIDRegex = new RegExp(/^[0-9]+$/);
const nonNumRegex = new RegExp(/[^0-9]/g);

export const onSubmit = (searchText: string, dispatch: Dispatch<any>): void => {
  if (searchText.match(palletIDRegex)) {
    dispatch(
      getPalletInfo({
        palletIds: [Number.parseInt(searchText, 10)],
        isAllItems: true,
        isSummary: false
      })
    );
  }
};

export const showActivitySpinner = (
  configWaiting: boolean,
  getInfoWaiting: boolean,
  getInfoCompleted: boolean
) => (getInfoWaiting && configWaiting) || (getInfoWaiting) || (configWaiting && getInfoCompleted);

export const getPalletInfoHook = (
  getPalletInfoApi: AsyncState,
  dispatch: Dispatch<any>,
  setGetInfoComplete: React.Dispatch<React.SetStateAction<boolean>>
) => {
  // on api success
  if (!getPalletInfoApi.isWaiting && getPalletInfoApi.result) {
    if (getPalletInfoApi.result.status === 200) {
      const {
        id, createDate, expirationDate, items
      } = getPalletInfoApi.result.data.pallets[0];
      const palletItems = items.map((item: PalletItem) => ({
        ...item,
        quantity: item.quantity || 0,
        newQuantity: item.quantity || 0,
        deleted: false,
        added: false
      }));
      const palletDetails: Pallet = {
        palletInfo: {
          id,
          createDate,
          expirationDate
        },
        items: palletItems
      };
      dispatch(setupPallet(palletDetails));
      setGetInfoComplete(true);
    } else if (getPalletInfoApi.result.status === 204) {
      Toast.show({
        type: 'error',
        text1: strings('LOCATION.PALLET_NOT_FOUND'),
        visibilityTime: 3000,
        position: 'bottom'
      });
    }
  }

  // on api error
  if (!getPalletInfoApi.isWaiting && getPalletInfoApi.error) {
    Toast.show({
      type: 'error',
      text1: strings('PALLET.PALLET_DETAILS_ERROR'),
      text2: strings('GENERICS.TRY_AGAIN'),
      visibilityTime: 4000,
      position: 'bottom'
    });
  }
};

export const getPalletConfigHook = (
  getPalletConfigApi: AsyncState,
  dispatch: Dispatch<any>,
  setConfigComplete: React.Dispatch<React.SetStateAction<boolean>>,
  backupCategories: string
) => {
  // on api success
  if (!getPalletConfigApi.isWaiting && getPalletConfigApi.result) {
    const { perishableCategories } = getPalletConfigApi.result.data;
    dispatch(setPerishableCategories(perishableCategories));
    dispatch({ type: GET_PALLET_CONFIG.RESET });
    setConfigComplete(true);
  }
  // on api error
  if (getPalletConfigApi.error) {
    const backupPerishableCategories = backupCategories.split(',').map(Number);
    dispatch(setPerishableCategories(backupPerishableCategories));
    dispatch({ type: GET_PALLET_CONFIG.RESET });
    setConfigComplete(true);
  }
};

export const PalletManagementScreen = (
  props: PalletManagementProps
): JSX.Element => {
  const {
    useEffectHook,
    searchText,
    setSearchText,
    configComplete,
    setConfigComplete,
    getInfoComplete,
    setGetInfoComplete,
    navigation,
    route,
    dispatch,
    getPalletInfoApi,
    getPalletConfigApi,
    userConfig
  } = props;

  let scannedSubscription: EmitterSubscription;

  const resetSearchText = () => setSearchText('');

  // Resets Get PalletInfo api state when navigating off-screen
  useEffectHook(() => {
    navigation.addListener('blur', () => {
      if (getPalletInfoApi.value) {
        dispatch({ type: GET_PALLET_INFO.RESET });
        resetSearchText();
      }
    });
  }, [getPalletInfoApi]);

  // Scanner listener
  useEffectHook(() => {
    dispatch(getPalletConfig());
    scannedSubscription = barcodeEmitter.addListener('scanned', scan => {
      if (navigation.isFocused()) {
        validateSession(navigation, route.name).then(() => {
          trackEvent('pallet_managment_scanned', {
            barcode: scan.value,
            type: scan.type
          });
          setSearchText(scan.value);
          dispatch(
            getPalletInfo({ palletIds: [scan.value], isAllItems: true, isSummary: false })
          );
        });
      }
    });
    return () => {
      scannedSubscription.remove();
    };
  }, []);

  // Get Pallet Info Api
  useEffectHook(() => {
    if (navigation.isFocused()) {
      getPalletInfoHook(getPalletInfoApi, dispatch, setGetInfoComplete);
    }
  }, [getPalletInfoApi]);

  //GetPalletConfig API
  useEffectHook(() => {
    if (navigation.isFocused()) {
      getPalletConfigHook(getPalletConfigApi, dispatch, setConfigComplete, userConfig.backupCategories);
    }
  }, [getPalletConfigApi]);

  useEffectHook(() => {
    if (navigation.isFocused()) {
      if (configComplete && getInfoComplete) {
        navigation.navigate('ManagePallet');
        setGetInfoComplete(false);
      }
    }
  }, [configComplete, getInfoComplete]);

  if (showActivitySpinner(getPalletConfigApi.isWaiting, getPalletInfoApi.isWaiting, getInfoComplete)) {
    return (
      <ActivityIndicator
        animating={showActivitySpinner(getPalletConfigApi.isWaiting, getPalletInfoApi.isWaiting, getInfoComplete)}
        hidesWhenStopped
        color={COLOR.MAIN_THEME_COLOR}
        size="large"
        style={styles.activityIndicator}
      />
    );
  }

  return (
    <View style={styles.scanContainer}>
      <TouchableOpacity onPress={() => openCamera()}>
        <Icon size={100} name="barcode-scan" color={COLOR.BLACK} />
      </TouchableOpacity>
      <View style={styles.scanText}>
        <Text>{strings('PALLET.SCAN_PALLET')}</Text>
      </View>
      <View style={styles.orText}>
        <Text>{strings('GENERICS.OR')}</Text>
      </View>
      <View style={styles.textView}>
        <TextInput
          value={searchText}
          onChangeText={(text: string) => setSearchText(text.replace(nonNumRegex, ''))}
          style={styles.textInput}
          keyboardType="numeric"
          placeholder={strings('PALLET.ENTER_PALLET_ID')}
          onSubmitEditing={() => onSubmit(searchText, dispatch)}
        />
      </View>
    </View>
  );
};

const PalletManagement = (): JSX.Element => {
  const getPalletInfoApi = useTypedSelector(state => state.async.getPalletInfo);
  const getPalletConfigApi = useTypedSelector(state => state.async.getPalletConfig);
  const userConfig = useTypedSelector(state => state.User.configs);
  const [configComplete, setConfigComplete] = useState(false);
  const [getInfoComplete, setGetInfoComplete] = useState(false);
  const [searchText, setSearchText] = useState('');
  const navigation = useNavigation();
  const route = useRoute();
  const dispatch = useDispatch();
  return (
    <PalletManagementScreen
      useEffectHook={useEffect}
      searchText={searchText}
      setSearchText={setSearchText}
      configComplete={configComplete}
      setConfigComplete={setConfigComplete}
      getInfoComplete={getInfoComplete}
      setGetInfoComplete={setGetInfoComplete}
      getPalletInfoApi={getPalletInfoApi}
      getPalletConfigApi={getPalletConfigApi}
      navigation={navigation}
      route={route}
      dispatch={dispatch}
      userConfig={userConfig}
    />
  );
};

export default PalletManagement;
