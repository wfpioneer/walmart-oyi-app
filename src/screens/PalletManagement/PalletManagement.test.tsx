import React from 'react';
import {
  NavigationContainer,
  NavigationContext,
  NavigationProp,
  RouteProp
} from '@react-navigation/native';
import ShallowRenderer from 'react-test-renderer/shallow';
import Toast from 'react-native-toast-message';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react-native';
import store from '../../state/index';
import { AsyncState } from '../../models/AsyncState';
import PalletManagement, {
  PalletManagementScreen,
  getPalletDetailsApiHook,
  navigateToPalletManageHook,
  setPerishableCategoriesHook
} from './PalletManagement';
import { mockConfig } from '../../mockData/mockConfig';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
      isFocused: jest.fn().mockReturnValue(true),
      goBack: jest.fn(),
      addListener: jest.fn()
    }),
    useRoute: () => ({
      key: 'test',
      name: 'Pallet Management Screen'
    })
  };
});

describe('PalletManagementScreen', () => {
  const defaultAsyncState: AsyncState = {
    isWaiting: false,
    value: null,
    error: null,
    result: null
  };

  const navigationProp: NavigationProp<any> = {
    addListener: jest.fn(),
    canGoBack: jest.fn(),
    getParent: jest.fn(),
    getId: jest.fn(),
    getState: jest.fn(),
    dispatch: jest.fn(),
    goBack: jest.fn(),
    isFocused: jest.fn(() => true),
    removeListener: jest.fn(),
    reset: jest.fn(),
    setOptions: jest.fn(),
    setParams: jest.fn(),
    navigate: jest.fn()
  };

  const routeProp: RouteProp<any, string> = {
    key: 'test',
    name: 'test'
  };
  const renderer = ShallowRenderer.createRenderer();

  describe('Tests rendering the PalletManagement Screen', () => {
    const actualNav = jest.requireActual('@react-navigation/native');
    const navContextValue = {
      ...actualNav.navigation,
      isFocused: () => false
    };
    it('render screen with redux', () => {
      const component = (
        <Provider store={store}>
          <NavigationContainer>
            <NavigationContext.Provider value={navContextValue}>
              <PalletManagement />
            </NavigationContext.Provider>
          </NavigationContainer>
        </Provider>
      );
      const { toJSON } = render(component);
      expect(toJSON()).toMatchSnapshot();
    });

    it('Renders the PalletManagement default ', () => {
      renderer.render(
        <PalletManagementScreen
          useEffectHook={jest.fn()}
          getPalletDetailsApi={defaultAsyncState}
          getInfoComplete={false}
          setGetInfoComplete={jest.fn()}
          navigation={navigationProp}
          dispatch={jest.fn()}
          route={routeProp}
          userConfig={mockConfig}
          isManualScanEnabled={true}
          trackEventCall={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
  describe('Tests rendering Get Pallet Info Api responses', () => {
    it('Renders Loading indicator if get pallet info waiting for an api response', () => {
      const palletDetailsIsWaiting: AsyncState = {
        ...defaultAsyncState,
        isWaiting: true
      };
      renderer.render(
        <PalletManagementScreen
          useEffectHook={jest.fn()}
          getPalletDetailsApi={palletDetailsIsWaiting}
          getInfoComplete={false}
          setGetInfoComplete={jest.fn()}
          navigation={navigationProp}
          dispatch={jest.fn()}
          route={routeProp}
          userConfig={mockConfig}
          isManualScanEnabled={true}
          trackEventCall={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });

  describe('Tests functions for hooks', () => {
    it('testgetPalletDetailsHook', async () => {
      const dispatch = jest.fn();
      const setGetInfoComplete = jest.fn();
      const successAsyncState = {
        ...defaultAsyncState,
        result: {
          status: 200,
          data: {
            pallets: [
              {
                id: 22,
                createDate: '2021-07-28T13:41:23.000Z',
                expirationDate: '04/13/2023 ',
                items: [
                  {
                    itemNbr: 210433,
                    itemDesc: '泰国金柚4.3KG',
                    price: 129,
                    upcNbr: '211149400000',
                    quantity: 14,
                    categoryNbr: 56
                  }
                ]
              }
            ]
          }
        }
      };
      const successNoPalletAsyncState = {
        ...defaultAsyncState,
        result: {
          status: 204
        }
      };
      const failureAsyncState = {
        ...defaultAsyncState,
        error: 'test'
      };
      getPalletDetailsApiHook(successAsyncState, dispatch, setGetInfoComplete);
      expect(dispatch).toHaveBeenCalled();
      expect(setGetInfoComplete).toHaveBeenCalled();

      getPalletDetailsApiHook(
        successNoPalletAsyncState,
        dispatch,
        setGetInfoComplete
      );
      expect(Toast.show).toBeCalledTimes(1);

      // @ts-expect-error mockReset is accessible during run time
      Toast.show.mockReset();
      getPalletDetailsApiHook(failureAsyncState, dispatch, setGetInfoComplete);
      expect(Toast.show).toBeCalledTimes(1);
    });
    it('Tests navigateToPalletManage Hook', () => {
      const mockNavigateCallback = jest.fn();
      const mockSetGetInfoComplete = jest.fn();
      navigateToPalletManageHook(
        false,
        navigationProp,
        jest.fn(),
        mockSetGetInfoComplete
      );
      expect(mockNavigateCallback).not.toBeCalled();
      navigateToPalletManageHook(
        true,
        navigationProp,
        jest.fn(),
        mockSetGetInfoComplete
      );
      expect(mockSetGetInfoComplete).toBeCalledWith(false);
    });
    it('Tests set perishable categories Hook', () => {
      const mockTrackEventCall = jest.fn();
      setPerishableCategoriesHook(
        '',
        navigationProp,
        mockTrackEventCall,
        jest.fn(),
        routeProp
      );
    });
  });
});
