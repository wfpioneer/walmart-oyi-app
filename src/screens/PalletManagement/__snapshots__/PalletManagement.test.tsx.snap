// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PalletManagementScreen Tests rendering Get Pallet Info Api responses Renders Loading indicator if get pallet info waiting for an api response 1`] = `
<ActivityIndicator
  animating={true}
  color="#0071e9"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;

exports[`PalletManagementScreen Tests rendering the PalletManagement Screen Renders the PalletManagement default  1`] = `
<ActivityIndicator
  animating={true}
  color="#0071e9"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;
